common/ustr.cpp

```diff
@@ -461,6 +461,57 @@ U32String convertUtf8ToUtf32(const String &str) {
 	return u32str;
 }
 
+// This is a quick and dirty converter.
```

Comment:
1. Original version is more effective. This version features `buffer = (char)(...) + buffer;` pattern, which causes unnecessary memory copying and reallocations, while original version works with raw bytes.
2. Original version had an option for performing strict/nonstrict conversion for 0xD800...0xDFFF interval.
3. Original version returned a result code of conversion. This version just insert '�' on characters that does not fit in 4 bytes and does not inform caller on any errors.

I guess, I can add those as "//TODO" since I already listed it?

Reply:
Yes, please add these points as a comment

Reply:
Added at 15df085197b6705cd02e4c5a9902d6934fdd3ac9.
Also reviewed convertUtf8ToUtf32() and wrote TODOs as well.